share/cmake/modules/FindOpenEXR.cmake

```diff
@@ -0,0 +1,249 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright Contributors to the OpenColorIO Project.
+#
+# Locate or install OpenEXR.
+#
+# Variables defined by this module:
+#   OpenEXR_FOUND - If FALSE, do not try to link to OpenEXR
+#   OpenEXR_LIBRARY - OpenEXR library to link to
+#   OpenEXR_INCLUDE_DIR - Where to find OpenEXR headers
+#   OpenEXR_VERSION - The version of the library
+#
+# Imported targets defined by this module, if found:
+#   OpenEXR::Iex
+#   OpenEXR::IexConfig
+#   OpenEXR::IlmThread
+#   OpenEXR::IlmThreadConfig
+#   OpenEXR::OpenEXR
+#   OpenEXR::OpenEXRConfig
+#   OpenEXR::OpenEXRCore
+#   OpenEXR::OpenEXRUtil
+#
+# By default, the dynamic libraries of OpenEXR will be found. To find the
+# static ones instead, you must set the OpenEXR_STATIC_LIBRARY variable to
+# TRUE before calling find_package(OpenEXR ...).
+#
+# If OpenEXR is not installed in a standard path, you can use the
+# OpenEXR_ROOT variable to tell CMake where to find it. If it is not found
+# and OCIO_INSTALL_EXT_PACKAGES is set to MISSING or ALL, OpenEXR will be
+# downloaded, built, and statically-linked into libOpenColorIO at build time.
```

Comment:
Good point Doug, I wonder what we should do in case OIIO is not found and we don't find OpenEXR on the system. Currently it will use ExternalProject as mentioned to bring it to the build, unless `OCIO_INSTALL_EXT_PACKAGES` is set to `NONE`. Should we change that to only build it if set to `ALL`? Alternatively use another CMake `option` to explicitly ask for `OpenEXR` or `OIIO` or `NONE` for the apps image backend?

Reply:
@doug-walker Maybe I misinterpreted your comment and you were more concerned with the `and statically-linked into libOpenColorIO at build time` part, which I should remove?

Reply:
@remia , yes, I was actually thinking about the linking part.  But your question about CMake is a good one too.  These would be great questions to ask during the TSC meeting on Monday.

Reply:
I think it's a nice feature to auto download EXR, like some of the other dependencies.  But I don't see why we would ever want to link it into the OCIO library since OCIO core does not depend on it, only the apps.

Reply:
@doug-walker I updated the comment as it was indeed incorrect and probably a left over from copy & paste.